platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/AssertionConsumerService.java

```diff
@@ -274,6 +274,21 @@ public Response processSamlResponse(org.opensaml.saml.saml2.core.Response samlRe
                         relayUri.getQuery(),
                         relayUri.getFragment());
             }
+            //this avoids the user ever being redirected to the built in login page if they've already
+            //logged in via the IDP.
+            if ((relayUri.getPath()
+                    .equals("/login") || relayUri.getPath()
+                    .equals("/login/")) && (relayUri.getQuery() != null && relayUri.getQuery()
+                    .contains("prevurl"))) {
+                relayUri = new URI(relayUri.getScheme(),
+                        relayUri.getUserInfo(),
+                        relayUri.getHost(),
+                        relayUri.getPort(),
+                        relayUri.getQuery()
+                                .replace("prevurl=", ""),
```

Comment:
If the path is /login or /login/" then remove the "prevurl" query param. That sounds right. 
Is the IDP path "/login" configurable. If it is can we pull the path from the configuration?
Also, Paths class will account for trailing slash:
` Paths.get("/login").equals(Paths.get("/login/")) == true`